pkg/virt-manifest/rest/manifest.go

```diff
@@ -0,0 +1,106 @@
+package rest
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+	ghodss_yaml "github.com/ghodss/yaml"
+	"k8s.io/client-go/pkg/util/yaml"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap"
+	"kubevirt.io/kubevirt/pkg/virt-manifest"
+)
+
+const (
+	MIME_YAML = "application/yaml"
+)
+
+type manifest struct {
+	libvirtUri    string
+	libvirtUser   string
+	libvirtPasswd string
+}
+
+type entityYAMLAccess struct {
+	ContentType string
+}
+
+// Read isn't currently used in this implementation
+func (e entityYAMLAccess) Read(req *restful.Request, v interface{}) error {
+	return fmt.Errorf("not implemented")
+}
+
+// Write marshals the value to YAML
+func (e entityYAMLAccess) Write(resp *restful.Response, status int, v interface{}) error {
+	data, err := ghodss_yaml.Marshal(v)
+	if err != nil {
+		return err
+	}
+	resp.WriteHeader(status)
+	resp.Write(data)
+	return nil
+}
+
+func ManifestService(libvirtUri string, libvirtUser string, libvirtPasswd string) (*restful.WebService, error) {
+	ws := new(restful.WebService)
+	m := manifest{
+		libvirtUri:    libvirtUri,
+		libvirtUser:   libvirtUser,
+		libvirtPasswd: libvirtPasswd}
+	restful.RegisterEntityAccessor(MIME_YAML, entityYAMLAccess{})
```

Comment:
The issue is that only JSON and XML are defined by default. If we do not explicitly define YAML, then if a client sends "Accept: application/yaml" it will receive an HTTP 406 error.